src/panels/config/helpers/dialog-helper-detail.ts

```diff
@@ -57,7 +57,7 @@ export class DialogHelperDetail extends LitElement {
 
```

Comment:
Here `_item` is typed as `Helper` (a union of the different types).  This is what the form elements expect to be passed.

But the `create...` functions expect the "`MutableParams`" types to be passed, which are not compatible with the `Helper` types, hence the error.

`_item` is actually set through the custom event value, but the code that fires that inside the forms is incredibly confusing.  I have no idea how to fix this because it just seems like a bug to me that might be causing problems somewhere?  There are lots of helper issues to pick from.

Reply:
I'll take a look

Reply:
Thanks.  Ultimately, `HELPERS` needs to be typed to avoid the reduction to `never`, but the problem becomes choosing a type for the 2nd argument:

```ts
type HelperCreators = {
  [domain in HelperDomain]: (hass: HomeAssistant, item: any) => Promise<Helper>;
};
```

Reply:
I pushed the type for `HELPERS` with `any` as a temporary fix with a comment about the mismatch.  I need the type predicate utility from this PR for other PRs, so maybe this issue can be logged and punted for now?